src/util.ts

```diff
@@ -133,6 +145,107 @@ export function getVersionFromFileContent(
   return version.toString();
 }
 
+function parseJavaVersionFile(content: string): string | null {
+  const javaVersionRegExp = /(?<version>(?<=(^|\s|\-))(\d+\S*))(\s|$)/;
+  const fileContent = content.match(javaVersionRegExp)?.groups?.version
+    ? (content.match(javaVersionRegExp)?.groups?.version as string)
+    : '';
+  if (!fileContent) {
+    return null;
+  }
+
+  return fileContent;
+}
+
+function parsePomXmlFile(xmlFileAsString: string): string | null {
+  const xmlDoc = create(xmlFileAsString);
+  const versionDefinitionTypes = [
+    getByMavenProperties,
+    getBySpringBootSpecification,
+    getByMavenCompilerPluginConfig
+  ];
+
+  for (const definitionType of versionDefinitionTypes) {
+    const version = definitionType(xmlDoc);
+
+    if (version !== null) {
+      return version;
+    }
+  }
+
+  return null;
+}
+
+function getByMavenProperties(xmlDoc: XMLBuilder): string | null {
+  const possibleTagsRegex = [
```

Comment:
These are not regular expressions, just tags. So please update the variable name. I suggest `versionTags`.